dartagnan/src/main/java/com/dat3m/dartagnan/parsers/program/visitors/spirv/ProgramBuilderSpv.java

```diff
@@ -591,5 +607,7 @@ public void setThreadGrid(List<Integer> threadGrid) {
             throw new ParsingException("Thread grid dimensions must be less than 128");
         }
         this.threadGrid = threadGrid;
+        BuiltIn builtIn = (BuiltIn) this.helperDecorations.getDecoration(BUILT_IN);
+        builtIn.setDimension(threadGrid.get(0), threadGrid.get(1), 1);
```

Comment:
Why is the last parameter always 1?

Reply:
Same as above.

Reply:
I see that you set BuildIn dimensions here so that we can distinguish the situation when config is not set explicitly and we use default (1,1,1). 

I would suggest to keep it simple and use whatever config we have after parsing headers, either default (1,1,1) or whatever is explicitly set by a user. Otherwise, in case of missing annotations, we can get weird situations where thread groupId doesn't agree with workgroup size.

And if dimensions are constant, then we can set them as final attributes in the BuiltIn constructor.